edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -28,7 +30,13 @@ public CertificateRenewal(EdgeHubCertificates certificates, ILogger logger)
             TimeSpan timeToExpire = certificates.ServerCertificate.NotAfter - DateTime.UtcNow;
             if (timeToExpire > TimeBuffer)
             {
-                var renewAfter = timeToExpire - TimeBuffer;
+                // Clamp the renew time to TimeSpan.FromMilliseconds(Int32.MaxValue)
+                // This is the maximum value for the timer (~24 days)
+                // Math.Min unfortunately doesn't work with TimeSpans so we need to do the check manually
+                TimeSpan withBuffer = timeToExpire - TimeBuffer;
+                TimeSpan renewAfter = withBuffer > MaxRenewAfter
+                    ? MaxRenewAfter
+                    : withBuffer;
```

Comment:
This forces us to restart every ~24 days, even if the cert renewal is longer - in the callback, should we check to see if we should wait longer and reset the timer?